Zend/zend_call_stack.c

```diff
@@ -450,12 +452,49 @@ static bool zend_call_stack_get_openbsd(zend_call_stack *stack)
 	return true;
 }
 #else
-static bool zend_call_stack_get_openbsd(zend_call_stack *stack)
+static bool zend_call_stack_get_openbsd_pthread(zend_call_stack *stack)
 {
 	return false;
 }
 #endif /* defined(HAVE_PTHREAD_STACKSEG_NP) */
 
+static bool zend_call_stack_get_openbsd_vm(zend_call_stack *stack)
+{
+	void *stack_base;
+	struct rlimit rlim;
+
+
+	if (getrlimit(RLIMIT_STACK, &rlim) != 0) {
+		return false;
+	}
+
+	if (rlim.rlim_cur == RLIM_INFINITY) {
+		return false;
+	}
+
+	// arch dependent top user's stack
+	stack->base = USRSTACK - rlim.rlim_cur;
```

Comment:
On my machine with OpenBSD 7.3 the actual stack base as reported by `procmap` is randomized, and the value of USRSTACK does not match. Maybe we could use sysctl() like pthread does?

Reply:
I ve looked at first at sysctl and did not find anything relevant (a way may have existed before tough), however there is a hope to achieve it I ll have a better look later today.